projects/canopy/src/lib/modal/modal-trigger/modal-trigger.directive.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+
+import { map } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+import { LgModalService } from '../modal.service';
+
+@Directive({
+  selector: '[lgModalTrigger]',
+})
+export class LgModalTriggerDirective implements OnInit, OnDestroy {
+  private allowFocusOnModalTrigger: boolean;
+  private subscription: Subscription;
+  @Input() lgModalTrigger: string;
+  @Output() clicked: EventEmitter<void> = new EventEmitter();
+
+  @HostListener('click') openModal(): void {
+    this.allowFocusOnModalTrigger = true;
+    this.modalService.open(this.lgModalTrigger);
+    this.clicked.emit();
+  }
+
+  constructor(private el: ElementRef, private modalService: LgModalService) {}
+
+  ngOnInit(): void {
+    this.subscription = this.modalService
+      .isOpen$(this.lgModalTrigger)
+      .pipe(
+        map((isOpen) => {
+          if (!isOpen && this.allowFocusOnModalTrigger) {
+            const triggerEl = this.el.nativeElement as HTMLElement;
```

Comment:
Minor, but can you drop the assignment?
```suggestion
            (this.el.nativeElement as HTMLElement).focus();
```

Reply:
Updated 👍